test/test_viscous.py

```diff
@@ -0,0 +1,520 @@
+"""Test the viscous fluid helper functions."""
+
+__copyright__ = """
+Copyright (C) 2021 University of Illinois Board of Trustees
+"""
+
+__license__ = """
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+"""
+
+import numpy as np
+import numpy.random
+import numpy.linalg as la  # noqa
+import pyopencl.clmath  # noqa
+import logging
+import pytest  # noqa
+
+from pytools.obj_array import make_obj_array
+from meshmode.dof_array import thaw
+from meshmode.mesh import BTAG_ALL
+import grudge.op as op
+from grudge.eager import (
+    EagerDGDiscretization,
+    interior_trace_pair
+)
+from meshmode.array_context import (  # noqa
+    pytest_generate_tests_for_pyopencl_array_context
+    as pytest_generate_tests)
+
+from mirgecom.fluid import make_conserved
+from mirgecom.transport import (
+    SimpleTransport,
+    PowerLawTransport
+)
+from mirgecom.eos import IdealSingleGas
+
+logger = logging.getLogger(__name__)
+
+
+def dont_test_actx_power(actx_factory):
+    """Test power of DOFArrays and the likes."""
+    actx = actx_factory()
+    dim = 3
+    nel_1d = 5
+    from meshmode.mesh.generation import generate_regular_rect_mesh
+    mesh = generate_regular_rect_mesh(
+        a=(1.0,) * dim, b=(2.0,) * dim, n=(nel_1d,) * dim
+    )
+    order = 1
+    discr = EagerDGDiscretization(actx, mesh, order=order)
+    nodes = thaw(actx, discr.nodes())
+
+    actx.np.power(nodes, .5)
+    actx.np.power(nodes[0], .5)
+    actx.np.power(nodes[0][0], .5)
```

Comment:
Just a reminder that we have no `actx.np.power` routine, which is used by the power-law transport model.  Since then, I've been advised to try the `**` operator - just haven't gotten around to doing that as it is not a part of this change set.